File: leetcode/palindrome_number/c/solve.cpp
#include <iostream>
#include <string>

class Solution {
public:
    bool isPalindrome(int x) {
        std::string xStr = std::to_string(x);
        int lengthOfStr = xStr.length();

        for (int i = 0, r = lengthOfStr - 1; i < lengthOfStr; i++, r--) {
            if (xStr[i] != xStr[r]) return false;
        }
        return true;
    }
};


int main() {
    Solution s = Solution();
    // test
    std::cout << s.isPalindrome(121) << std::endl;
    std::cout << s.isPalindrome(-121) << std::endl;
    std::cout << s.isPalindrome(101) << std::endl;
    return 0;
}

File: leetcode/palindrome_number/python/solve.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        str_x = str(x)
        reverse = str_x[::-1]
        if str_x == reverse:
            return True
        return False

if __name__ == "__main__":
    s = Solution()
    # test
    print(s.isPalindrome(121))
    print(s.isPalindrome(-121))
    print(s.isPalindrome(10))
    print(s.isPalindrome(101))